packages/common/src/common.ts

```diff
@@ -505,8 +534,13 @@ export class Common extends EventEmitter {
    * @param td Total difficulty
    *    * @returns The value requested or `BigInt(0)` if not found
    */
-  paramByBlock(topic: string, name: string, blockNumber: BigIntLike, td?: BigIntLike): bigint {
-    const hardfork = this.getHardforkByBlockNumber(blockNumber, td)
+  paramByBlock(
+    topic: string,
+    name: string,
+    blockNumber: BigIntLike,
+    { td, timestamp }: { td?: BigIntLike; timestamp?: BigIntLike }
```

Comment:
I didn't know you could declare parameters like this. Will have to experiment with it and see if it makes sense.